src/rust/engine/options/src/lib.rs

```diff
@@ -186,15 +186,15 @@ pub(crate) trait OptionsSource {
     /// Get the dict option identified by `id` from this source.
     /// Errors when this source has an option value for `id` but that value is not a dict.
     ///
-    fn get_dict(&self, id: &OptionId) -> Result<Option<Vec<DictEdit>>, String>;
+    fn get_dict(&self, id: &OptionId) -> Result<Option<DictEdit>, String>;
 }
 
-#[derive(Copy, Clone, Debug, Ord, PartialOrd, Eq, PartialEq)]
+#[derive(Clone, Debug, Ord, PartialOrd, Eq, PartialEq)]
```

Comment:
Ord will do the right thing here: order by increasing declaration order, and within Config, lexicographically by field, hence `ordinal`.

Reply:
It falls out of the documentation for the generated Ord, but can't hurt to add one, done.

Reply:
I was thinking the test would be catching if someone jumps in and changes the enum (eg sorts the variants alphabetically) they may not realise that the variant and field order is specifically meaningful, rather than _any_ order being fine (and just being necessary to allow use in BTreeMap or similar).

Reply:
Yep, a test is a good idea, added one.